ComponentKit/Core/Scope/CKComponentScopeRootFactory.h

```diff
@@ -0,0 +1,17 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <Foundation/Foundation.h>
+
+@protocol CKComponentStateListener;
+
+@class CKComponentScopeRoot;
+
+CKComponentScopeRoot *CKComponentScopeRootWithListener(id<CKComponentStateListener> listener);
```

Comment:
Yeah, the predicates we'd want to add would be used in addition to the current predicates for appearance and bounds animations.